core/src/main/scala/akka/persistence/cassandra/journal/CassandraDeletion.scala

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.persistence.cassandra.journal
+
+import akka.persistence.cassandra.query.EventsByPersistenceIdStage.Extractors
+import akka.persistence.cassandra.query.scaladsl.CassandraReadJournal
+import akka.persistence.cassandra.session.scaladsl.CassandraSession
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.Sink
+import com.datastax.driver.core.policies.{ LoggingRetryPolicy, RetryPolicy }
+import com.datastax.driver.core._
+import java.lang.{ Long => JLong }
+
+import scala.language.implicitConversions
+import akka.persistence.cassandra._
+import akka.Done
+import akka.event.LoggingAdapter
+import akka.persistence.cassandra.journal.CassandraDeletion.{ MessageId, PartitionInfo }
+
+import scala.collection.immutable
+import scala.concurrent.{ ExecutionContext, Future }
+
+object CassandraDeletion {
+  private case class PartitionInfo(partitionNr: Long, minSequenceNr: Long, maxSequenceNr: Long)
+  private case class MessageId(persistenceId: String, sequenceNr: Long)
+}
+
+trait CassandraDeletion extends CassandraStatements {
+  private[akka] val session: CassandraSession
+
+  private[akka] val log: LoggingAdapter
+  private[akka] def config: CassandraJournalConfig
+  private[akka] def queries: CassandraReadJournal
+  private[akka] implicit val ec: ExecutionContext
+  private[akka] implicit val materializer: ActorMaterializer
+
+  private[akka] val readRetryPolicy: RetryPolicy
+  private lazy val deleteRetryPolicy = new LoggingRetryPolicy(new FixedRetryPolicy(config.deleteRetries))
+
+  def preparedSelectDeletedTo = session.prepare(selectDeletedTo)
+    .map(_.setConsistencyLevel(config.readConsistency).setIdempotent(true).setRetryPolicy(readRetryPolicy))
+  def preparedSelectHighestSequenceNr = session.prepare(selectHighestSequenceNr)
+    .map(_.setConsistencyLevel(config.readConsistency).setIdempotent(true).setRetryPolicy(readRetryPolicy))
+  def preparedInsertDeletedTo = session.prepare(insertDeletedTo)
+    .map(_.setConsistencyLevel(config.writeConsistency).setIdempotent(true))
+  def preparedDeleteMessages = session.prepare(deleteMessages).map(_.setIdempotent(true))
+
+  def asyncDeleteMessagesTo(persistenceId: String, toSequenceNr: Long): Future[Unit] = {
```

Comment:
Not new code, just moved from CassandraJournal to make it usable without creating a Journal